driver/fvGFS/atmosphere.F90

```diff
@@ -2010,6 +2016,28 @@ subroutine adiabatic_init(zvir,nudge_dz,time)
      do_adiabatic_init = .false.
      call timing_off('adiabatic_init')
 
+ contains
+
+   subroutine adjust_tracer_mass()
+
+     if ( nq > Atm(mygrid)%flagstruct%nwat ) then
+       ! Adjust prognostic tracer mixing ratios for consistency with updated layer mass
+!$omp parallel do default (none) &
+!$omp              shared (jsc, jec, isc, iec, npz, nq, Atm, dp0, mygrid) &
+!$omp             private (i, j, k, n)
+       do n = Atm(mygrid)%flagstruct%nwat + 1, nq
+         do k = 1, npz
+           do j = jsc, jec
+             do i = isc, iec
+               Atm(mygrid)%q(i,j,k,n) = Atm(mygrid)%q(i,j,k,n) * (1. + wt * (1. - dp0(i,j,k) / Atm(mygrid)%delp(i,j,k)))
```

Comment:
Besides condensates, are there any other tracer types that shouldn't be scaled in this fashion - e.g. number concentrations?

Reply:
Any tracer type whose concentration is expressed as ratio of a given quantity to air mass will need to be adjusted.

Reply:
@rmontuoro I am wondering what happens to the prognostic subgrid scale turbulent kinetic energy "sgs_tke" which is also after nwat, are you also going to adjust it? Have you done testing to see if the number concentration fields or the tke field should be adjusted? Can you show some of your results and test configuration from this update? I am not sure if we have restriction that all the prognostic tracers after nwat are expressed as ratio to air mass.